backend/src/main/java/hu/flowacademy/meteo/controller/AirDataController.java

```diff
@@ -0,0 +1,26 @@
+package hu.flowacademy.meteo.controller;
+
+import hu.flowacademy.meteo.dto.AirDataDto;
+import hu.flowacademy.meteo.dto.BetweenDatesDto;
+import hu.flowacademy.meteo.dto.MeasurmentDto;
+import hu.flowacademy.meteo.service.MeasurmentService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequiredArgsConstructor
+public class AirDataController {
+
+    private final MeasurmentService measurmentService;
+
+    @PostMapping("air")
+    public List<AirDataDto> getAllBetweenDates
+            (@RequestBody BetweenDatesDto betweenDatesDto) {
+        return measurmentService.getAllBetweenDates(betweenDatesDto.getStartDate(),
+                betweenDatesDto.getEndDate(), betweenDatesDto.getType(), betweenDatesDto.getId()).stream()
```

Comment:
Én az ID-t nem a BetweenDatesDto-ból venném, hanem az URL-be raknám, mint erőforrás azonosító. API felhasználóként nekem valami ilyen GET query lenne logikus és kézenfekvő a 123-as állomás levegő adataihoz: /api/stations/123/air?startDate=1624984804&endDate=1624984813&type=daily

Reply:
De nem látom a big picture-t, szóval most lehet egy nagyon nagy hülyeséget mondtam.

Reply:
kb így voolt a mai napig, csak többen mondták hogy request body-val jobb lenne. ezért ma átírtam arra....

Reply:
Ok, lehet, ha megmondja valaki, hogy miért lenne jobb.

Reply:
Ha csak azért mert fáj stringet összefűzni egy objectből, akkor mondom, hogy az axios tud ilyet, a paramsban kb ugyanaz menne, mint a post bodyban:

```
axios.get('/stations/123/air', {
    params: {
      startDate: 1624984804,
      endDate: 1624984813,
      type: 'daily'
    }
  })
```

Reply:
Mehet paramban is, de en alapvetoen nem szeretek komplexebb queryparamokat kipakolni a path-re, es bodybol kenyelmesebb kiszedni/kezelni, foleg ha meg ennel is komplexebbre no.

Reply:
Mindkettő élhető szerintem is, talán én itt a get-et és a fent Roland által említett példát választottam volna
(/api/stations/123/air?startDate=1624984804&endDate=1624984813&type=daily), de simán érvelhető a másik is. Ha hozzá veszzük, hogy több releváns adat esetén még a paging is bejöhet, talán a path param-os egy konvencionálisabb.